src/network/crypto.rs

```diff
@@ -6,44 +6,54 @@ use tracing::{info, instrument, trace};
-                _ => panic!("unknown key for version {version}")
-            };
-            BASE64_STANDARD.decode(b64).unwrap()
-        })
-        .to_vec();
-
-    info!(version, "found initial decryption key");
+pub fn lookup_initial_key(initial_keys: &HashMap<u32, Vec<u8>>, bytes: &[u8]) -> Result<Vec<u8>> {
+    let version = u32::from_be_bytes(
+        bytes
+            .get(..4)
+            .context("failed to get version's bytes")?
+            .try_into()?,
+    ) ^ 0x9D74C714;
 
-    key
+    {
+        if let Some(x) = initial_keys.get(&version).map(|k| k.to_vec()) {
+            info!(version, "found initial decryption key");
+            return Ok(x);
+        }
+    }
+
+    'a: {
+        let b64 = match version {
```

Comment:
I'd move these into files and use `include_str!` or, better, `include_bytes!`, which makes the Base64 decoding unnecessary.